pkg/controller/servicebinding/secret.go

```diff
@@ -10,98 +12,108 @@ import (
 	"k8s.io/client-go/dynamic"
 
 	"github.com/redhat-developer/service-binding-operator/pkg/converter"
-	"github.com/redhat-developer/service-binding-operator/pkg/log"
 )
 
-// secret represents the data collected by this operator, and later handled as a secret.
-type secret struct {
-	logger *log.Log          // logger instance
-	client dynamic.Interface // Kubernetes API client
-	ns     string
-	name   string
-}
-
 // buildResourceClient creates a resource client to handle corev1/secret resource.
-func (s *secret) buildResourceClient() dynamic.ResourceInterface {
+func buildResourceClient(client dynamic.Interface, namespace string) dynamic.ResourceInterface {
 	gvr := corev1.SchemeGroupVersion.WithResource(secretResource)
-	return s.client.Resource(gvr).Namespace(s.ns)
+	return client.Resource(gvr).Namespace(namespace)
+}
+
+// name is a utility function to get a checksum of the resource data
+func name(data map[string][]byte) (string, error) {
+	keys := make([]string, 0, len(data))
+	for k := range data {
+		keys = append(keys, k)
+	}
+	sort.Strings(keys)
+	hash := sha1.New()
+	for _, k := range keys {
+		_, err := hash.Write([]byte(k))
+		if err != nil {
+			return "", err
+		}
+		_, err = hash.Write(data[k])
+		if err != nil {
+			return "", err
+		}
+	}
+	return hex.EncodeToString(hash.Sum(nil)), nil
 }
 
-// createOrUpdate will take informed payload and either create a new secret or update an existing
-// one. It can return error when Kubernetes client does.
-func (s *secret) createOrUpdate(payload map[string][]byte, ownerReference metav1.OwnerReference) (*unstructured.Unstructured, error) {
-	logger := s.logger.WithValues("Namespace", s.ns, "Name", s.name)
-	secretObj := &corev1.Secret{
+// write will take informed payload and create a new secret
+// It can return error when Kubernetes client does.
+func write(
```

Comment:
Before this symbol was scoped by the `secret` type, but now it doesn't and the function (which is available to the entire `servicebinding` package) has a very opaque name. Please change it to `writeServiceBindingSecret(client, ns, prefix, payload, ownerReference)` or something more meaningful instead.